src/Chapter2.hs

```diff
@@ -728,7 +758,8 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
+smartReplicate (x:xs) = replicate x x ++ (smartReplicate xs)
```

Comment:
Great! 👍🏼 

You implemented this function using manual recursion over the list.

If you want to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests 🙂 But you still need to use `replicate` 👍🏻